Typos (#2096)
javascript-tutorial/en.javascript.info: 2-ui/99-ui-misc/03-event-loop/article.md

## Event Loop

> The *event loop* concept is very simple. There's an endless loop, when the JavaScript engine waits for tasks, executes them and then sleeps, waiting for more tasks.

Reviewer: I think it should be 'where' instead of 'when'.

when the JavaScript engine waits for tasks ->  where the JavaScript engine waits for tasks

So far, quite simple, right?

Two more details:
1. Rendering never happens while the engine executes a task. It doesn't matter if the task takes a long time. Changes to the DOM are painted only after the task is complete.

> 2. If a task takes too long, the browser can't do other tasks, process user events, so after a time it raises an alert like "Page Unresponsive" suggesting killing the task with the whole page. That happens when there are a lot of complex calculations or a programming error leading to infinite loop.

Reviewer: ```suggestion
2. If a task takes too long, the browser can't do other tasks, such as processing user events. So after a time, it raises an alert like "Page Unresponsive", suggesting killing the task with the whole page. That happens when there are a lot of complex calculations or a programming error leading to an infinite loop.
```

Another benefit of splitting heavy tasks for browser scripts is that we can show progress indication.

> Usually the browser renders after the currently running code is complete. It doesn't matter if the task takes a long time. Changes to DOM are painted only after the task is finished.

Reviewer: Maybe...

```suggestion
As mentioned earlier, changes to DOM are painted only after the currently running task is completed, irrespective of how long it takes.
```
